native/dlaqr5.go

```diff
@@ -0,0 +1,616 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dlaqr5 performs a single small-bulge multi-shift QR sweep on an isolated
+// block of a Hessenberg matrix.
+//
+// wantt and wantz determine whether the quasi-triangular Schur factor and the
+// orthogonal Schur factor, respectively, will be computed.
+//
+// kacc22 specifies the computation mode of far-from-diagonal orthogonal
+// updates. Permitted values are:
+//  0: Dlaqr5 will not accumulate reflections and will not use matrix-matrix
+//     multiply to update far-from-diagonal matrix entries.
+//  1: Dlaqr5 will accumulate reflections and use matrix-matrix multiply to
+//     update far-from-diagonal matrix entries.
+//  2: Dlaqr5 will accumulate reflections, use matrix-matrix multiply to update
+//     far-from-diagonal matrix entries, and take advantage of 2×2 block
+//     structure during matrix multiplies.
+// For other values of kacc2 Dlaqr5 will panic.
+//
+// n is the order of the Hessenberg matrix H.
+//
+// ktop and kbot are indices of the first and last row and column of an isolated
```

Comment:
I thought about this too but in the end decided that keeping kbot inclusive is the least change both in meaning of the parameters and in the actual implementation. The same with mbot. I will keep it as is but feel free to bug me if you oppose.